kvbc/src/kvbc_app_filter/kvbc_app_filter.cpp

```diff
@@ -388,7 +388,11 @@ std::optional<uint64_t> KvbAppFilter::getNextEventGroupId(std::shared_ptr<EventG
   // Vice versa is true if event_group_id_batch is populated with only private event group ids
   // Note: each private_event_group_ids and public_event_group_ids hold a sorted list of global event group ids at all
   // times
-  if (!public_event_group_ids.empty() &&
+  bool is_lte_batch_size = (!public_event_group_ids.empty() && !private_event_group_ids.empty() &&
+                            public_event_group_ids.size() + private_event_group_ids.size() <= kBatchSize)
+                               ? true
+                               : false;
+  if (!public_event_group_ids.empty() && !is_lte_batch_size &&
```

Comment:
Instead of `is_lte_batch_size`, could we use `public_event_group_ids.size() >= kBatchSize`? Similar for private events?
IMHO, this makes it more readable because you only consider public events if all public events available can fill the whole batch + other conditions.

Reply:
Thanks, yes it does. Updated.